### ISSUE TITLE
Problematic login-logout flow

### ISSUE THREAD

**Issue by HarelM** [openstreetmap/openstreetmap-website#3236]
The following flow creates a problematic experience:
1. Login to UserA
2. Go to user settings
3. Logout
4. Login with user B
5. --> After login the redirection is going to UserA settings page which is unauthorized (403 error) and the screen doesn't go to the regular openstreetmap.org page, but says 403 error...

**Comment by tomhughes** [comment]
I agree it's not ideal but that's a pretty unusual set of actions and it would be relatively hard to fix without breaking lots of more useful flows.

To be honest I would have expect the problem to occur after step 3 in that list - does the logout not try and send you back to the settings page that is no longer available?

**Comment by tomhughes** [comment]
Thinking about it I guess it might be possible to drop the referer from the logout link on any page which requires authentication. I need to think a bit more about it though...

**Comment by Nitishsuman1** [comment]
After logout, session data is cleared.

When User B logs in, they’re redirected to / or their dashboard — not User A’s settings.

Unauthorized redirects are avoided or blocked with checks.

**Comment by AntonKhorev** [comment]
Was this issue about `/user/:display_name/account` pages? Account/profile/preferences settings pages no longer have usernames in their paths, so the scenario as described in this issue is not going to happen.

**Comment by HarelM** [comment]
I don't think this is a very interesting flow to be honest, just wanted to report it in case there's an interest to solve it.
If this is no longer a possible flow, feel free to close the issue.
